pytorch_lightning/core/lightning.py

```diff
@@ -1839,11 +1844,34 @@ def to_onnx(
 
         input_sample = self._apply_batch_transfer_handler(input_sample)
 
+        if not isinstance(input_sample, (tuple, list)):
+            input_sample = (input_sample, )
+
         if "example_outputs" not in kwargs:
             self.eval()
-            kwargs["example_outputs"] = self(input_sample)
+            kwargs["example_outputs"] = self(*input_sample)
 
         torch.onnx.export(self, input_sample, file_path, **kwargs)
+
+        def default_model_check_fn(p, inp, torch_outs):
```

Comment:
can you move this outside the function as a protected `staticmethod`? Avoiding potential pickle erros here...